core/sqf/src/seatrans/tm/hbasetmlib2/javaobjectinterfacetm.cpp

```diff
@@ -421,21 +422,32 @@ bool  JavaObjectInterfaceTM::getExceptionDetails(JNIEnv *jenv)
        _tlp_error_msg = error_msg;
        return false;
    }
+   error_msg = new std::string("");
+   appendExceptionMessages(jenv, a_exception, error_msg);
+   *error_msg += "\n";
+   _tlp_error_msg = error_msg;
+   return true;
+}
+
+void JavaObjectInterfaceTM::appendExceptionMessages(JNIEnv *jenv, jthrowable a_exception, std::string *error_msg)
```

Comment:
If we don't expect error_msg to be null, it might be better to code this as a reference parameter rather than a pointer.